apps/server/src/shared/domain/entity/user.entity.ts

```diff
@@ -156,11 +156,38 @@ export class User extends BaseEntityWithTimestamps implements EntityWithSchool {
 			permissions = [...permissions, ...rolePermissions];
 		});
 
-		const uniquePermissions = [...new Set(permissions)];
+		const setOfPermission = this.checkSchoolPermissions(permissions, roles);
+
+		const uniquePermissions = [...setOfPermission];
 
 		return uniquePermissions;
 	}
 
+	private checkSchoolPermissions(permissions: string[], roles: Role[]) {
+		const setOfPermission = new Set(permissions);
+		const schoolPermissions = this.school.permissions;
+
+		if (this.school.permissions) {
```

Comment:
we should keep it consistent with the logic from /me endpoint
in case we have a user with both admin and teacher role, for example..... 
https://github.com/hpi-schul-cloud/schulcloud-server/pull/4875/files#diff-aee2907d716d0b306a2f45ea7a06b3991cdd5e1623c61ac6291990cdbac312d7R14